tests/data/examples/pubmed/generate_data.py

```diff
@@ -0,0 +1,185 @@
+    """
+    Retrieve PubMed papers by query.
+
+    Args:
+        query (str):
+            The search query for PubMed.
+        retmax (int):
+            The maximum number of results to return.
+
+    Returns:
+        List[Dict[str, Any]]:
+            A list of PubMed articles.
+
+    """
+    # capture ids which match the query
+    id_list = Entrez.read(
+        Entrez.esearch(
+            db="pubmed", sort="relevance", retmax=str(retmax), retmode="xml", term=query
+        )
+    )["IdList"]
+
+    # fetch article data using the article ids
+    return Entrez.read(Entrez.efetch(db="pubmed", retmode="xml", id=",".join(id_list)))[
+        "PubmedArticle"
+    ]
+
+
+def get_article_date(article: Dict[Any, Any]) -> str:
+    """
+    Extract the publication date of a PubMed article.
+
+    Args:
+        article (Dict[Any, Any]):
+            The PubMed article in dictionary format.
+
+    Returns:
+        str:
+            The publication date in YYYY-MM-DD format, or None if not available.
+
+    """
+    try:
+        # form a string data from the various separated values
+        return (
+            f"{article['MedlineCitation']['Article']['ArticleDate'][0]['Year']}-"
+            f"{article['MedlineCitation']['Article']['ArticleDate'][0]['Month']}-"
+            f"{article['MedlineCitation']['Article']['ArticleDate'][0]['Day']}"
+        )
+    except Exception:
+        return None
+
+
+def get_abstract(paper: Dict[Any, Any]) -> str:
+    """
+    Retrieve the abstract of a PubMed article.
+
+    Args:
+        paper (Dict[Any, Any]):
+            The PubMed article in dictionary format.
+
+    Returns:
+        str:
+            The abstract text of the article, or blank string if not available.
+
+    """
+
+    if "Abstract" in paper["MedlineCitation"]["Article"]:
+        if isinstance(
+            abstract := paper["MedlineCitation"]["Article"]["Abstract"]["AbstractText"],
+            list,
+        ):
+            # if we have the abstract, transform it to a single string
+            return " ".join(abstract)
+
+    return ""
+
+
+def extract_github_links(text: str) -> List[str]:
+    """
+    Extract GitHub links from a given text.
+
+    Args:
+        text (str):
+            The text to search for GitHub links.
+
+    Returns:
+        List[str]:
+            A list of GitHub links found in the text in the format
+            "https://github.com/<some text>/<some text>",
+            without anything after the second slash.
+
+    """
+
+    # Regex to find GitHub links in the specified format
+    github_links = re.findall(r"https?://github\.com/[^/]+/[^/\s\),.]+", text)
+
+    # Removing anything after the second slash
+    cleaned_links = []
+    for link in github_links:
+        match = re.match(r"(https?://github\.com/[^/]+/[^/]+)", link)
+        if match:
+            cleaned_links.append(match.group(1))
+
+    return cleaned_links
+
+
+def is_github_link_valid(link: str) -> bool:
```

Comment:
In the parquet file it seems that  `isValid = False` is still in the dataframe. For example,

 `10238	37304664	2023-05-26	Palmprint recognition based on gating mechanis...	Zhang, Xu, Jin, Qi, Yang, Bai	https://github.com/Ywatery/GLnet	False`. 
 
 Was the parquet file updated? This could also be a problem on my end. Let me know what you think!

Reply:
Great point! It looks like I didn't export properly, I'll revise and add this in a further commit.